tests/test_metadata.py

```diff
@@ -76,6 +88,81 @@ def test_preauthencoding(self):
         # Checking for Pre-Auth-Encoding generated is correct.
         self.assertEqual(self.pae, envelope_obj.pae)
 
+    def test_sign(self):
+        """Test payload signing of envelope"""
+
+        # Create an Evnvelope with no signatures
+        envelope_dict = copy.deepcopy(self.envelope_dict)
+        envelope_dict["signatures"] = []
+        envelope_obj = Envelope.from_dict(envelope_dict)
+
+        for (_, key) in self.key_dicts:
+            # Test with invalid signer
+            with self.assertRaises(TypeError):
+                envelope_obj.sign(key)
+
+            # Test for invalid scheme
+            valid_scheme = key["scheme"]
+            key["scheme"] = "invalid_scheme"
+            signer = SSlibSigner(key)
+            with self.assertRaises((FormatError, UnsupportedAlgorithmError)):
+                envelope_obj.sign(signer)
+
+            # Sign the payload
+            key["scheme"] = valid_scheme
+            signer = SSlibSigner(key)
+            envelope_obj.sign(signer)
+
+        # Tests for signatures created
+        self.assertEqual(len(self.key_dicts), len(envelope_obj.signatures))
+        for signature in envelope_obj.signatures:
+            self.assertIsInstance(signature, Signature)
+
+    def test_verify(self):
+        "Test payload verification of envelope"
+
+        # Create an Evnvelope with no signatures
+        envelope_dict = copy.deepcopy(self.envelope_dict)
+        envelope_dict["signatures"] = []
+        envelope_obj = Envelope.from_dict(envelope_dict)
+
+        # sign payload of envelope
+        for (_, key) in self.key_dicts:
+            signer = SSlibSigner(key)
+            envelope_obj.sign(signer)
+
+        # check for signatures
+        self.assertEqual(len(self.key_dicts), len(envelope_obj.signatures))
+
+        keys_list = [
+            (name, SSlibKey.from_securesystemslib_key(key))
+            for (name, key) in self.key_dicts
+        ]
```

Comment:
It's a little ambiguous when you have a complicated one-liner.
Maybe it's just better to use a `for` loop here.